premake/wxwidgets.lua

```diff
@@ -132,7 +132,9 @@ end
 
 local function wxSetTargetDirectory(arch, build)
   -- Target directory
-  if (is_msvc) then
+  if arch == "MacOsx" then
```

Comment:
`MacOsx` is not an architecture, but a platform or system. You are dropping the compiler-specific part from the path, but even on MacOS you could use different compilers like `gcc` or `clang`.